script.js

```diff
@@ -1,82 +1,217 @@
-  episodeSelectElement.className = "col-3 m-3 p-2 form-control";
-  episodeSelectElement.id = "selectEpisode";
-  searchSection.append(
-    showSelectElement,
-    episodeSelectElement
+// Make section of filtering show
+function makeFilterSection() {
+  let filterSectionElement = document.createElement("section");
+  filterSectionElement.id = "filterSection";
+  filterSectionElement.style.display = "block";
+  filterSectionElement.className = "px-3 ";
+  let filterDivElement = document.createElement("div");
+  filterDivElement.className = "row p-3 text-start  ";
+  let filteringLabelElement = document.createElement("h6");
+  filteringLabelElement.className = "col-lg-1 col-md-2 col-4 m-1 p-2";
+  filteringLabelElement.textContent = `Filtering for `;
+  let parentOfInputElement = document.createElement("div");
+  parentOfInputElement.className = "col-md-3 col-7 p-0";
+  let filterInputElement = document.createElement("input");
+  filterInputElement.id = "filterKey";
+  filterInputElement.className = "form-control p-1 m-1";
+  filterInputElement.placeholder = "Enter to search show";
+  parentOfInputElement.append(filterInputElement);
+  let filterFoundLabelElement = document.createElement("h6");
+  filterFoundLabelElement.id = "countOfShows";
+  filterFoundLabelElement.className = " col-md-3 col-7 p-2 pl-lg-3 ";
+  let parentOfSelectElement = document.createElement("div");
+  parentOfSelectElement.className = "col-md-3 col-11 ml-1 p-0";
+  let filterSelectElement = document.createElement("select");
+  filterSelectElement.id = "filteredShows";
+  filterSelectElement.className = "form-control p-1 m-1";
+  parentOfSelectElement.append(filterSelectElement);
+  filterDivElement.append(
+    filteringLabelElement,
+    parentOfInputElement,
+    filterFoundLabelElement,
+    parentOfSelectElement
   );
+  filterSectionElement.append(filterDivElement);
+  const rootElement = document.getElementById("root");
+  rootElement.append(filterSectionElement);
+  fillShowsDropDown();
+  filterSelectElement.addEventListener("change", function () {
+    const rootElem = document.getElementById("root");
+    const showSectionElement = document.getElementById("displayShows");
+    rootElem.removeChild(showSectionElement);
+    let selectedShow = getSelectedShow(this.value);
+    makeShowSection(selectedShow);
+  });
+
+  function getSelectedShow(selectedId) {
+    return getAllShows().filter((show) => show.id == selectedId);
+  }
+  filterInputElement.addEventListener("input", function () {
+    const rootElem = document.getElementById("root");
+    const showSectionElement = document.getElementById("displayShows");
+    rootElem.removeChild(showSectionElement);
+    let filteredShows = fillShowsDropDown();
+    makeShowSection(filteredShows);
+  });
 }
 
-function fillSelectByShows(selectedShow) {
-  let searchSection = document.getElementById("searchSection");
-  let SelectElement = document.getElementById("showSelect");
```

Comment:
Even though it does not affect your code in any way, i think it would be better to keep naming variables in `camelCase` as you did with most variables in your project. 
Also found a very short article about it: 
https://techterms.com/definition/camelcase#:~:text=Software%20developers%20often%20use%20camelCase,easier%20to%20read%20than%20myonemethod

Reply:
Thank you for giving attention to details.